### ISSUE TITLE
Request for copy of the source code for v4.0.2

### ISSUE THREAD

**Issue by hchendow** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#1643]
I'm trying to get my full framework API app working with Identity Server 4 using the default at+jwt token header instead of changing it to use the jwt header which I know works. I was hoping to get a copy of the source code for v4.0.2 and trying to see if I can make some changes in order to support what I'm trying to do. Any help is greatly appreciated!

**Comment by mafurman** [comment]
Could you elaborate a bit more on what you're trying to do?

You should be able to view the source code for the 4.x branch here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/tree/dev4x

**Comment by hchendow** [comment]
Thanks for sharing that link.

Yeah I can elaborate some more. 

Relevant NuGet Packages:
- Identity Model v1.9.2
- IdentityServer3.AccessTokenValidation v2.14
- Microsoft.Owin v4.1.1
- System.IdentityModel.Tokens.Jwt v4.0.2

I have a .NET 4.6.1 API that is using the OWIN/Katana setup in order to perform Authentication using the bearer tokens. I'm using the code below in the API's OWIN Startup.cs file.
       
app.UseIdentityServerBearerTokenAuthentication(new IdentityServerBearerTokenAuthenticationOptions()
  {
      Authority = "...",
      AuthenticationType = "Bearer",
      RequiredScopes = new[] { "..." },
      ValidationMode = ValidationMode.Local
  
  });

in order to authenticate any bearer tokens that hit the API. The tokens should be coming from an Identity Server 4 setup we have with Admin UI set up. With the current configuration of IDS4 we don't use the options below to allow for 'JWT' tokens.

 services.AddIdentityServer(options =>
            {
                options.AccessTokenJwtType = "JWT";
                options.EmitLegacyResourceAudienceClaim = true; //Allows JWT header instead of just jwt+at
            })

That said, when anything tries to authenticate to the API with it's current packages and setup I get this error: 
'IDX10703: Unable to decode the 'header': ... SecurityTokenException: IDX10702: Jwt header type specified, must be 'JWT' or 'http://openid.net/specs/jwt/1.0'.  Type received: 'at+jwt'.

I know that this is because the System.IdentityModel.Tokens.Jwt doesn't accept a token with the header type of at+jwt or that it can't decode it and that's what I wanted to see if I could do myself. I was thinking if I could get the source code for 4.0.2 then I could try updating it to decode the tokens while supporting the current framework my application is running on.

I cannot upgrade to .NET Core (though I would love to) due to dependencies on vendor packages that don't support .NET Core. I hope this explains enough.

**Comment by hchendow** [comment]
Is there anyway for me to compile that code locally and use it with my application while still having the private key signature? I'm running into the infamous "manifest doesn't match assembly reference" error if I don't use the SNK to sign the assembly.

**Comment by mafurman** [comment]
I'm not aware of a way for you to compile it locally while still maintaining the signature on it.

I'm also having trouble understanding why you're setting the following property:

`options.AccessTokenJwtType = "JWT";`

It appears that in IdentityServer this value defaults to "at+jwt"?
https://identityserver4.readthedocs.io/en/latest/reference/options.html

**Comment by hchendow** [comment]
@mafurman The reason I needed to change the type is because the IdentityServer3.AccessTokenValidation package depends on System.IdentityModel.Tokens.Jwt v4.0.2 (Anything 5+ doesn't work) and that package doesn't support "at+jwt" tokens type headers. That is why in order to support the v4.0.2 package, we set the options in Identity Server to use "JWT" type header instead of the default "at+jwt" type header.

That said, my team asked me to look at seeing if we can update the System.IdentityModel.Tokens.Jwt v4.0.2 so that it supported the "at+jwt" token instead of going backwards and trying to support the full framework APIs I'm working with. That is why I've been trying to update the System.IdentityModel.Tokens.Jwt package to support the "at+jwt"
